pipeline/splitters.go

```diff
@@ -97,6 +102,87 @@ func (t *TokenSplitter) FindRecord(buf []byte) (bytesRead int, record []byte) {
 	return bytesRead, buf[:bytesRead]
 }
 
+type MultilineSplitter struct {
+	delimiter  *regexp.Regexp
+	multiline  *regexp.Regexp
+	captureLen int
+}
+
+type MultilineSplitterConfig struct {
+	Delimiter string
+	Multiline string
+}
+
+func (r *MultilineSplitter) ConfigStruct() interface{} {
+	return &MultilineSplitterConfig{
+		Delimiter: "\n",
+	}
+}
+
+func (r *MultilineSplitter) Init(config interface{}) error {
+	conf := config.(*MultilineSplitterConfig)
+	var err error
+	if r.delimiter, err = regexp.Compile(conf.Delimiter); err != nil {
+		return err
+	}
+	if r.delimiter.NumSubexp() > 1 {
+		return fmt.Errorf("regex must not contain more than one capture group: %s",
+			conf.Delimiter)
+	}
```

Comment:
Not sure why this constraint is here. In the RegexSplitter, the capture group specifies which part of the regex match, if any, should be included in the record that is emitted. AFAICT you're always throwing the delimiter away, so this has no impact.

Ultimately, it would probably be better to match the behaviour of the RegexSplitter, to enable people to keep the delimiter in the record if desired.